node-wasm/src/node.rs

```diff
@@ -38,141 +37,226 @@ pub struct WasmNodeConfig {
     pub bootnodes: Vec<String>,
 }
 
-#[wasm_bindgen(js_class = Node)]
-impl WasmNode {
-    /// Create a new Lumina node.
+#[wasm_bindgen]
+struct NodeDriver {
+    _worker: SharedWorker,
+    _onerror_callback: Closure<dyn Fn(MessageEvent)>,
+    channel: WorkerClient,
+}
+
+#[wasm_bindgen]
+impl NodeDriver {
+    /// Create a new connection to a Lumina node in a Shared Worker.
+    /// Note that single Shared Worker can be accessed from multiple tabs, so Lumina may already
+    /// be running be running, before `NodeDriver::start` call.
     #[wasm_bindgen(constructor)]
-    pub async fn new(config: WasmNodeConfig) -> Result<WasmNode> {
-        let config = config.into_node_config().await?;
+    pub async fn new() -> Result<NodeDriver> {
+        let worker = spawn_worker(LUMINA_SHARED_WORKER_NAME, "/wasm/lumina_node_wasm.js")?;
```

Comment:
I'm not sure I follow - two above linked properties would allow us to get a URL of the page that the user is on, but what we need here is wasm entrypoint location.
Even though we're essentially loading the same wasm module as is running again, I don't see a way to get it's location programatically at runtime.

Reply:
if we cannot get this programatically, then I think we should accept it as an argument to avoid hardcode in library code

Reply:
switched to running a worker from a static script, since with blob url the url was unique for each lumina instance and resulted in many shared workers instead of a single. This is completely hidden from user, we get the worker path programatically